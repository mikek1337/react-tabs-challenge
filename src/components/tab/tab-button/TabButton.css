.tab-button{
    display: flex;
    flex-direction: row;
    background-color: #262626;
    height: 60px;
    overflow-x: auto;
   
}
.tab-title-container{
    display: flex;
    justify-content: center;
    align-items: center;

    text-align: center;
    width: 100%;
    height: 100%;
}
.tab-title-container span{
    color: #9CA3AF;
}

.tab-title-container:hover{
    background-color: #3B3B3B;
    cursor: pointer;
}

.tab-title-container.active{
    background-color: #0284C7;
  
}
.tab-title-container.active span{
    color: #FFFFFF;
}