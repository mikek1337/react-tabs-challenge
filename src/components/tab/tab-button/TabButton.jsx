import './TabButton.css'
export default function TabButton({tabs, activeTabIndex, onTabClick}){
    return(
        <div className='tab-button'>
            {
                tabs?.map((tab, index)=>(
                    <div className={'tab-title-container '+ (activeTabIndex === index?"active":"")} onClick={()=>{onTabClick && onTabClick(index)}} key={tab.id}>
                         <span>{tab.tabTitle}</span>
                    </div>
                ))
            }
        </div>
    )
}