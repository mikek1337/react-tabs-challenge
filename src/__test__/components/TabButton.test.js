import { fireEvent, render } from "@testing-library/react";
import TabButton from "../../components/tab/tab-button/TabButton";

describe("TabButton", () => {
  const TestTabData = [
    {
      id: 1,
      tabTitle: "Tab 1",
      title: "Title 1",
    },
    {
      id: 2,
      tabTitle: "Tab 2",
      title: "Title 2",
    },
    {
      id: 3,
      tabTitle: "Tab 3",
      title: "Title 3",
    },
    {
      id: 4,
      tabTitle: "Tab 4",
      title: "Title 4",
    },
  ];
  it("should render", () => {
    const onTabClick = jest.fn();
    const { container } = render(
      <TabButton
        tabs={TestTabData}
        activeTabIndex={0}
        onTabClick={onTabClick}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render with active tab", () => {
   const mockParent = {
    activeTabIndex:0,
    onTabClick: jest.fn((index)=>{mockParent.activeTabIndex = index})
   }
    const { container } = render(
      <TabButton
        tabs={TestTabData}
        {...mockParent}
      />
    );
    
    fireEvent.click(container.querySelectorAll(".tab-title-container")[1]);
    
    expect(mockParent.activeTabIndex).toBe(1);
  });
});
