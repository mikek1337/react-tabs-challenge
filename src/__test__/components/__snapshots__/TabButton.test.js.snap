// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton should render 1`] = `
<div>
  <div
    class="tab-button"
  >
    <div
      class="tab-title-container active"
    >
      <span>
        Tab 1
      </span>
    </div>
    <div
      class="tab-title-container "
    >
      <span>
        Tab 2
      </span>
    </div>
    <div
      class="tab-title-container "
    >
      <span>
        Tab 3
      </span>
    </div>
    <div
      class="tab-title-container "
    >
      <span>
        Tab 4
      </span>
    </div>
  </div>
</div>
`;
